applications/Preprocessor/lib/mesh/MeshImpl/Mesh.h

```diff
@@ -194,6 +195,30 @@ class Mesh {
                    : 0;
     }
 
+    /**
+     * Get the number of entities of a certain dimension.
+     *
+     * This includes the unused entities (i.e. entities without an attached
+     * element).
+     *
+     * Note the following equivalence should hold for 0 <= i <= dimension
+     * getNumberOfEntities(i) == getNumberOfEntities<i>(), but the latter is
+     * probably slightly faster.
+     *
+     * @param entityDim The dimension of the entity, must be <= dimension
+     * @return The number entities
+     */
+    std::size_t getNumberOfEntities(std::size_t entityDim) const {
+        if (entityDim == dimension) {
+            return elementsList.size();
+        } else {
+            logger.assert_debug(
+                entityDim < dimension,
+                "Asking for entities with dimension higher than the mesh.");
+            return getNumberOfEntities(entityDim, itag<dimension - 1>{});
```

Comment:
I do not understand this.

Reply:
I've extracted the algorithm for this dynamic access into `TemplateArray`. I hope it is more clear in that context.